test/plugins/http2/client.spec.js

```diff
@@ -0,0 +1,806 @@
+
+              req.end()
+            })
+          })
+        })
+
+        it(`should support request configuration without a path and method`, done => {
+          const app = (stream, headers) => {
+            stream.respond({
+              ':status': 200
+            })
+            stream.end()
+          }
+
+          getPort().then(port => {
+            agent
+              .use(traces => {
+                expect(traces[0][0].meta).to.have.property('span.kind', 'client')
+                expect(traces[0][0].meta).to.have.property('http.url', `${protocol}://localhost:${port}/`)
+              })
+              .then(done)
+              .catch(done)
+
+            appListener = server(app, port, () => {
+              const client = http2
+                .connect(`${protocol}://localhost:${port}`)
+                .on('error', done)
+
+              const req = client.request({})
+              req.on('error', done)
+
+              req.end()
+            })
+          })
+        })
+
+        it('should support connect configuration with a URL object', done => {
+          const app = (stream, headers) => {
+            stream.respond({
+              ':status': 200
+            })
+            stream.end()
+          }
+
+          getPort().then(port => {
+            agent
+              .use(traces => {
+                expect(traces[0][0].meta).to.have.property('http.url', `${protocol}://localhost:${port}/user`)
+              })
+              .then(done)
+              .catch(done)
+
+            const uri = {
+              protocol: `${protocol}:`,
+              hostname: 'localhost',
+              port
+            }
+
+            appListener = server(app, port, () => {
+              const client = http2
+                .connect(uri)
+                .on('error', done)
+
+              const req = client.request({ ':path': '/user' })
+              req.on('error', done)
+
+              req.end()
+            })
+          })
+        })
+
+        it('should remove the query string from the URL', done => {
+          const app = (stream, headers) => {
+            stream.respond({
+              ':status': 200
+            })
+            stream.end()
+          }
+
+          getPort().then(port => {
+            agent
+              .use(traces => {
+                expect(traces[0][0].meta).to.have.property('http.url', `${protocol}://localhost:${port}/user`)
+              })
+              .then(done)
+              .catch(done)
+
+            appListener = server(app, port, () => {
+              const client = http2
+                .connect(`${protocol}://localhost:${port}`)
+                .on('error', done)
+
+              const req = client.request({ ':path': '/user?foo=bar' })
+              req.on('error', done)
+
+              req.end()
+            })
+          })
+        })
+
+        it('should support a string URL and an options object, with the string URL taking precedence', done => {
```

Comment:
These tests are failing due to http2's handling of overridable options: https://github.com/nodejs/node/blob/master/lib/internal/http2/core.js#L2788

When using the `net` (http) server, the `authority` configuration takes precedence over `options`. However, when using the `tls` (https) server, the options take precedence when creating the socket. Since this functionality is really only relevant to http2, I think we can just remove the affected tests.